Rakefile

```diff
@@ -14,66 +14,50 @@ task :generate_csv do
     "ocd-division/country:ug/" + parts.map { |type, value| [type, idify(value)].join(':') }.join('/')
   end
 
-  csv_url = 'https://docs.google.com/spreadsheets/d/1LetNFNq6ovg4bbq-Whze0Q06CYSpNqpRIzSUb9yVtw4/export?format=csv'
-
-  csv = CSV.parse(open(csv_url).read, headers: true, header_converters: :symbol)
-
-  ocd_ids = Set.new
-
-  csv.each do |r|
-    next unless r[:arearegion]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion]),
-      name: r[:arearegion]
-    }
-
-    next unless r[:areasub_region]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion], subregion: r[:areasub_region]),
-      name: r[:areasub_region]
-    }
-
-    next unless r[:areadistrict]
-    ocd_ids << {
-      id: id_for(region: r[:arearegion], subregion: r[:areasub_region], district: r[:areadistrict]),
-      name: r[:areadistrict]
-    }
-
-    next unless r[:areaconstituency]
-    ocd_ids << {
-      id: id_for(region:r[:arearegion], subregion: r[:areasub_region], district: r[:areadistrict], constituency: r[:areaconstituency]),
-      name: r[:areaconstituency]
-    }
+  def ocd_ids_from_csv(csv_data, mapping = {})
+    csv = CSV.parse(csv_data, headers: true, header_converters: :symbol)
+    ocd_ids = Set.new
+
+    csv.each do |r|
+      region = r[mapping.fetch(:region, :region)]
+      next unless region
+      ocd_ids << {
+        id: id_for(region: region),
+        name: region.strip
+      }
+
+      subregion = r[mapping.fetch(:subregion, :subregion)]
+      next unless subregion
+      ocd_ids << {
+        id: id_for(region: region, subregion: subregion),
+        name: subregion.strip
+      }
+
+      district = r[mapping.fetch(:district, :district)]
+      next unless district
+      ocd_ids << {
+        id: id_for(region: region, subregion: subregion, district: district),
+        name: district.strip
+      }
+
+      constituency = r[mapping.fetch(:constituency, :constituency)]
+      next unless constituency
+      next if ['-', '_'].include?(constituency)
```

Comment:
It's not immediately obvious to me what this is doing, and feels like it might be in the wrong place. (i.e. although this isn't quite a completely generic function, it's fairly close to being able to lift it and turn into a standalone class, whereas this doesn't seem like something that would be part of that — though tricky to know as I can't quite tell why it's here…)